resources/qml/Cura.qml

```diff
@@ -359,157 +143,199 @@ UM.MainWindow
                 {
-                    top: topbar.bottom
-                    bottom: parent.bottom
                     left: parent.left
-                    right: sidebar.left
-                }
+                    margins: UM.Theme.getSize("default_margin").width
 
-                MouseArea
-                {
-                    visible: UM.Controller.activeStage.mainComponent != ""
-                    anchors.fill: parent
-                    acceptedButtons: Qt.AllButtons
-                    onWheel: wheel.accepted = true
+                    bottom: parent.bottom
                 }
-
-                source: UM.Controller.activeStage.mainComponent
             }
 
-            Loader
+            ComboBox
             {
-                id: sidebar
-
-                property bool collapsed: false;
-                property var initialWidth: UM.Theme.getSize("sidebar").width;
-
-                function callExpandOrCollapse() {
-                    if (collapsed) {
-                        sidebar.visible = true;
-                        sidebar.initialWidth = UM.Theme.getSize("sidebar").width;
-                        viewportRect = Qt.rect(0, 0, (base.width - sidebar.width) / base.width, 1.0);
-                        expandSidebarAnimation.start();
-                    } else {
-                        viewportRect = Qt.rect(0, 0, 1, 1.0);
-                        collapseSidebarAnimation.start();
+                // This item contains the views selector, a combobox that is dynamically created from
+                // the list of available Views (packages that create different visualizations of the
+                // scene).
+                id: viewModeButton
+
+                anchors.left: viewOrientationControls.right
+                anchors.bottom: viewOrientationControls.bottom
+
+                style: UM.Theme.styles.combobox
+
+                model: UM.ViewModel { }
+                textRole: "name"
+
+                // update the model's active index
+                function updateItemActiveFlags ()
+                {
+                    currentIndex = getActiveIndex()
+                    for (var i = 0; i < model.rowCount(); i++)
+                    {
+                        model.getItem(i).active = (i == currentIndex)
                     }
-                    collapsed = !collapsed;
-                    UM.Preferences.setValue("cura/sidebar_collapsed", collapsed);
                 }
 
-                anchors
+                // get the index of the active model item on start
+                function getActiveIndex ()
                 {
-                    top: topbar.top
-                    bottom: parent.bottom
+                    for (var i = 0; i < model.rowCount(); i++)
+                    {
+                        if (model.getItem(i).active)
+                        {
+                            return i;
+                        }
+                    }
+                    return 0
```

Comment:
The combination of `updateItemActiveFlags` and `getActiveIndex` seems to be circularly dependent in a way. I understand that it works: Calling `updateItemActiveFlags()` causes it to flag only the first item as active that is active. But there should be an easier way. Just a loop that goes over them that runs until it encounters the first active item and makes the rest inactive.

Reply:
This is something that was just copy-pasted and I don't think it's important to change it now, since we are changing the whole behavior in this sprint.